### PR TITLE
feat: Fix portions of web functionality

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

Some features of the Flutter SDK that could work in Web weren't working.  While we still don't have full feature parity for Flutter Web, this should bring us closer.

In addition, we're adding web integration tests to the CI to ensure that we at least keep current levels of functionality moving forward.

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests 
- [ ] This pull request references a Github or JIRA issue

### CI Configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests

### CODE REVIEW

File: packages/datadog_flutter_plugin/lib/src/logs/ddlogs_web.dart [hunk 1]
```diff
@@ -84,7 +86,19 @@ class DdLogsWeb extends DdLogsPlatform {
...
+      error = JSError();
+      error.stack = convertWebStackTrace(errorStackTrace);
+      error.message = errorMessage;
+      error.name = errorKind ?? 'Error';
+    }
+    logger?.log(
+      message,
+      valueToJs(attributes, 'attributes'),
+      webLogLevel,
+      error,
```
Comment (0xnm) [inline packages/datadog_flutter_plugin/lib/src/logs/ddlogs_web.dart:100]: error can be uninitialized here if `if` block above is not executed? what will be the behavor in this case, will we have a crash?

File: packages/datadog_flutter_plugin/lib/src/web_helpers.dart [hunk 2]
```diff
@@ -52,3 +53,30 @@ dynamic valueToJs(Object? value, String parameterName) {
   throw ArgumentError(
       'Could not convert ${value.runtimeType} to javascript.', parameterName);
 }
+
+String? convertWebStackTrace(StackTrace? stackTrace) {
+  if (stackTrace == null) return null;
+
+  final dartLineRegex =
+      RegExp(r'(?<file>.+) (?<location>\d+:\d+)\s*(?<function>.+)');
```
Comment (0xnm) [inline packages/datadog_flutter_plugin/lib/src/web_helpers.dart]: should we extract it a static field or simply class property? it is same expression everytime, so maybe we can create the object once and compile the expession, no need to create another one with each method invocation.

File: packages/datadog_flutter_plugin/lib/src/logs/ddlogs_web.dart (hunk 1 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/lib/src/logs/ddlogs_web.dart:100]: Dart defaults to `null` on a nullable value, and `log` supports null errors so we should be good.

File: packages/datadog_flutter_plugin/lib/src/web_helpers.dart (hunk 2 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/lib/src/web_helpers.dart]: Yes, good call. I originally tried to make it `const` but apparently RegExp doesn't have a const constructor.

File: packages/datadog_flutter_plugin/lib/src/logs/ddlogs_web.dart (hunk 1 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/lib/src/logs/ddlogs_web.dart:100]: (And this case should be handled in the integration tests)

**Review by 0xnm — APPROVED** [review]

**Pull request merged**